### PR TITLE
Add jupyter releaser workflows

### PR DESCRIPTION
**Pull request by mahendrapaipuri**
Added [jupyter-releaser workflows](https://github.com/jupyter-server/jupyter_releaser/tree/main/.github/actions) and a `CHANGELOG.md` file generated by `github-activity`.

### CODE REVIEW

**Comment by jtpio** [comment]
Thanks @mahendrapaipuri!

Wondering if we could use the `python_packages` field of the releaser to specify all the Python packages in the monorepo?  Like the `jupyverse` project does here: https://github.com/jupyter-server/jupyverse/blob/b3c3589ef71d9b8088fb86cea3c92746ea05d49b/pyproject.toml#L144-L164

We could check if it is working by looking at the uploaded artifacts by the check release workflow on CI.

**Comment by mahendrapaipuri** [comment]
This is what exactly I am looking at. As we do not have a `pyproject.toml` on the root of the repo, we will have to add a `.jupyter-releaser.toml` for config. JupyterLab is using sort of a dummy [package name](https://github.com/jupyterlab/jupyterlab/blob/f40bb0a8d619c429d700c305bdb5dfa45bd91a52/package.json#L2C3-L5C15) for the root `package.json` and using [`build-before-npm`](https://github.com/jupyterlab/jupyterlab/blob/f40bb0a8d619c429d700c305bdb5dfa45bd91a52/pyproject.toml#L227-L230) hook to build the packages. We should be able to do the same right?

We need to add script to `package.json` to bump versions of all individual packages though.

**Comment by jtpio** [comment]
We could also have a `pyproject.toml` at the root level of the repo to hold the configuration I guess.

**Comment by mahendrapaipuri** [comment]
@jtpio I tried to integrate the `check-release` workflow but with some **strong** assumptions.

- Added a dummy python package in the top-level directory and version in `package.json`. We should make sure that the version in the top level directory should be in sync will all packages.
- We cannot make individual release to a package. Everytime we run release workflows, all package versions will be bumped. Given that they are all pure TS extensions, maybe we can live that?

If we can pass `--python-packages` arg at runtime, I think we can control the individual package release. But I guess that is not possible at the moment? Let me know, what do you think!

File: .github/workflows/publish-release.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+      branch:
+        description: "The target branch"
+        required: false
+      release_url:
+        description: "The URL of the draft GitHub release"
+        required: false
+      steps_to_skip:
+        description: "Comma separated list of steps to skip"
+        required: false
+        default: "ensure-sha"
```
Comment (jtpio) [inline .github/workflows/publish-release.yml:14]: We can remove this default. It's only needed in JupyterLab for now: https://github.com/jupyterlab/jupyterlab/issues/13442

```suggestion
```

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+[tool.hatch.version]
+source = "nodejs"
+
+[tool.hatch.metadata.hooks.nodejs]
+fields = ["description", "authors", "urls"]
+
+[tool.jupyter-releaser.options]
+version-cmd = "hatch version"
+python-packages = [
+    ".:jupyterlab_topbar_extensions",
```
Comment (jtpio) [inline pyproject.toml:21]: Does this create artifacts for the top-level package?

Wondering if we should remove it as we are only interested in releasing the individual extensions?

```suggestion
```

Comment (mahendrapaipuri) [reply pyproject.toml]: It does create the top-level package. Probably it was left after my local testing. Let's see if CI complains or not!

**Comment by cccs-nik** [comment]
Any updates on this @jtpio? I would like to use the new version of jupyterlab-topbar for JupyterLab 4.0 🙂

**Comment by jtpio** [comment]
I think we should be good to go.

Maybe we can grab the latest version of the workflows from the releaser repo to get the proper configuration for using the PyPI trusted publishers: https://github.com/jupyter-server/jupyter_releaser/blob/main/example-workflows/publish-release.yml

**Comment by jtpio** [comment]
Also it looks like we can version all the packages together for now, although in the long run it would be nice to allow versioning packages independently.

**Comment by mahendrapaipuri** [comment]
> Also it looks like we can version all the packages together for now, although in the long run it would be nice to allow versioning packages independently.

@jtpio Agree! Lets do a release for Jupyterlab 4.0 now and we will see how we can version packages independently!

**Comment by mahendrapaipuri** [comment]
@jtpio Just checking in to see if there is anything missing in this PR to get it merged and do a release?

**Comment by jtpio** [comment]
@mahendrapaipuri yes it should be good to go!

Merging now, will try to get a release out today. Thanks again!

**Comment by jtpio** [comment]
> Merging now, will try to get a release out today. Thanks again!

Just tried the prep release workflow specifying `1.0.0` as the version but it bumped to `0.6.1` instead:

image

Looks like there might be an issue when reading the new version after the bump.

**Pull request merged**